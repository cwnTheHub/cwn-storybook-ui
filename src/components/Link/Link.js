import React, { useState } from "react";
import { BsChevronDown } from "react-icons/bs";
import { useLocation } from "react-router-dom";
import {
  DropdownContainer,
  DropdownLink,
  DropdownMenu,
  LinkButton,
  LinkWithDecoration,
  LinkWithoutDecoration,
} from "./LinkStyles";

const Link = ({
  right,
  isButton,
  dropdownLink,
  theLink,
  icon,
  activeLink,
  onClick,
  onSelectActiveLink,
  isactive,
}) => {
  const { pathname } = useLocation();
  const [isDropdownClicked, setIsDropdownClicked] = useState(false);

  const handleClickOnDropdown = (ev) => {
    ev.preventDefault();
    setIsDropdownClicked(!isDropdownClicked);
  };

  const handleBlurOnDropdown = (ev) => {
    ev.preventDefault();
    setIsDropdownClicked(!isDropdownClicked);
  };

  if (dropdownLink) {
    return (
      <DropdownContainer data-testid="dropdown-container">
        <DropdownLink
          onClick={handleClickOnDropdown}
          data-testid="dropdown-link"
          open={isDropdownClicked}
          actived={isactive}
        >
          {icon || null}
          {activeLink?.title || theLink?.title}
          <BsChevronDown size={12} />
        </DropdownLink>
        {isDropdownClicked && (
          <DropdownMenu
            open={isDropdownClicked}
            right={right}
            data-testid="dropdown-menu"
            tabindex="0"
            onBlur={handleBlurOnDropdown}
          >
            {theLink?.categories?.map((link, index) => (
              <div key={index}>
                <LinkWithDecoration
                  to={link?.to}
                  actived={
                    link?.to?.toLowerCase() === pathname?.toLowerCase() ? 1 : 0
                  }
                  onClick={() => onSelectActiveLink(link)}
                >
                  {link?.title}
                </LinkWithDecoration>
                {index > 0 && index % 3 === 0 && <span>tst</span>}
              </div>
            ))}
          </DropdownMenu>
        )}
      </DropdownContainer>
    );
  }
  if (isButton) {
    return (
      <LinkButton
        to={theLink?.to}
        actived={theLink?.to?.toLowerCase() === pathname?.toLowerCase() ? 1 : 0}
        onClick={() => {
          onClick(theLink);
          onSelectActiveLink(theLink);
        }}
        data-testid="button-link"
      >
        {theLink?.title}
      </LinkButton>
    );
  }
  return (
    <LinkWithoutDecoration
      to={theLink?.to}
      actived={theLink?.to?.toLowerCase() === pathname?.toLowerCase() ? 1 : 0}
      onClick={() => {
        onClick(theLink);
        onSelectActiveLink(theLink);
      }}
      data-testid="normal-link"
    >
      {icon || null}
      <span>{theLink?.title}</span>
    </LinkWithoutDecoration>
  );
};

export default Link;
