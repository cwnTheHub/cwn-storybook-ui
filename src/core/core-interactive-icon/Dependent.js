import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import {
  colorGreyShark,
  colorNemetonPurple,
  colorWhite,
  colorAccessibleGreen,
} from "../core-colours/colours";
import { DependentIconSizeContext } from "../../util-helpers";

const positionStyles = ({ paragraphSize }) => {
  let top = 0;
  if (paragraphSize === "large") {
    top = "-4px";
  }

  return {
    position: "relative",
    top,
  };
};

export const StyledDependentSVG = styled.svg.attrs({
  "aria-hidden": true,
  focusable: false,
})(
  positionStyles,
  ({ paragraphSize }) => ({
    width: paragraphSize === "small" ? "1.25rem" : "1.5rem",
    height: paragraphSize === "small" ? "1.25rem" : "1.5rem",
  }),
  ({ color }) => {
    let fill;
    if (color === "greyShark") {
      fill = colorGreyShark;
    } else if (color === "white") {
      fill = colorWhite;
    } else if (color === "nemetonPurple") {
      fill = colorNemetonPurple;
    } else if (color === "accessibleGreen") {
      fill = colorAccessibleGreen;
    }
    return { fill };
  }
);

const Dependent = ({ children, ...rest }) => {
  return (
    <DependentIconSizeContext.Consumer>
      {({ paragraphSize }) => {
        return React.cloneElement(children, {
          paragraphSize,
          "data-testid": "dependentSvg",
          ...rest,
        });
      }}
    </DependentIconSizeContext.Consumer>
  );
};

Dependent.propTypes = {
  /**
   * @ignore
   */
  children: PropTypes.node.isRequired,
  color: PropTypes.oneOf([
    "greyShark",
    "white",
    "nemetonPurple",
    "accessibleGreen",
  ]),
};

Dependent.defaultProps = {
  color: "greyShark",
};
export default Dependent;
