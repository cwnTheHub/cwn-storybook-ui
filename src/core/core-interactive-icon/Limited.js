import React from "react";
import PropTypes from "prop-types";
import styled, { ThemeProvider } from "styled-components";
import StyledInteractiveIconSVG from "./shared/StyledInteractiveIconSVG";
import {
  colorShark,
  colorNemetonPurple,
  colorWhite,
  colorAccessibleGreen,
  colorCardinal,
} from "../core-colours/colours";
import animations from "./shared/animations";
import { responsiveProps } from "../../util-prop-types";
import { handleResponsiveStyles } from "../../util-helpers";

const iconSize = (props) =>
  handleResponsiveStyles({ size: props.size }, ({ size }) => ({
    width: size === 20 ? "1.25rem" : "1.5rem",
    height: size === 20 ? "1.25rem" : "1.5rem",
  }));

export const StyledLimitedInteractiveIconSVG = styled(StyledInteractiveIconSVG)(
  ({ animationDirection }) => ({
    transition: "transform 150ms ease-in-out",
    "@media (prefers-reduced-motion: reduce)": {
      transition: "none",
    },
    [":hover, :focus, :active"]: {
      transform: `translate${
        animationDirection === "up" || animationDirection === "down" ? "Y" : "X"
      }(${
        animationDirection === "up" || animationDirection === "left" ? "-" : ""
      }4px)`,
    },
  }),
  animations.reduceMotion,
  iconSize
);

const getTheme = (variant) => {
  if (variant === "basic") {
    return {
      backgroundColor: "transparent",
      iconColor: colorShark,
    };
  }
  if (variant === "alternative") {
    return {
      backgroundColor: "transparent",
      iconColor: colorNemetonPurple,
    };
  }
  if (variant === "inverted") {
    return {
      backgroundColor: "transparent",
      iconColor: colorWhite,
    };
  }
  if (variant === "error") {
    return {
      backgroundColor: "transparent",
      iconColor: colorCardinal,
    };
  }
  return {
    backgroundColor: "transparent",
    iconColor: colorAccessibleGreen,
  };
};

const Limited = ({ variant, children, size }) => (
  <ThemeProvider theme={getTheme(variant)} size={size}>
    {children}
  </ThemeProvider>
);

Limited.displayName = "Limited";

Limited.propTypes = {
  /**
   * The style.
   */
  variant: PropTypes.oneOf([
    "default",
    "basic",
    "alternative",
    "inverted",
    "error",
  ]),
  size: responsiveProps(PropTypes.oneOf([16, 24])),

  children: PropTypes.node.isRequired,
};

Limited.defaultProps = {
  variant: "default",
  size: 24,
};

export default Limited;
