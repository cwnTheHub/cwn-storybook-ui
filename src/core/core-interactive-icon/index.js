export * from "./svgs";
export { default as IconButton } from "./IconButton";
export {default as NavButton} from "./NavButton";
export {default as Limited} from "./Limited";