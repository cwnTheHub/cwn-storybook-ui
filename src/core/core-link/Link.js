import React, { useCallback } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import {
  colorGreyShark,
  colorWhite,
  colorGainsboro,
  colorGreyRaven,
} from "../core-colours/colours";
import { links } from "../../shared-styles";
import { safeRest } from "../../util-helpers";
import { componentWithName } from "../../util-prop-types";
import { withForwardedRef } from "../../shared-hocs";
import { warn } from "../../utils/warn";

const base = {
  ...links.focusOutline,
  "&:link,&:visited": {
    color: colorGreyShark,
    textDecoration: "underline",
  },
  "&:hover": {
    textDecoration: "none",
  },
  "& svg": {},
};

const states = ({ invert }) => {
  return {
    "&:active": {
      color: invert && colorGainsboro,
      backgroundColor: invert ? "rgba(0,0,0,0.4)" : colorGainsboro,
      borderRadius: "0.25rem",
      padding: "0.125rem",
      margin: "-0.125rem",
      textDecoration: "underline",
    },
    "&:focus": {
      border: `0.125rem solid ${invert ? colorWhite : colorGreyRaven}`,
      padding: "0.125rem",
      margin: "-0.25rem", // (border + padding) * -1
      borderRadius: "0.25rem",
      outline: "none",
    },
  };
};
const StyledLink = styled.a(
  base,
  {
    "& svg": {
      transition: "transform 150ms ease-in-out",
    },
    "&:hover svg": {
      transform: "scale(1.1, 1.1)",
    },
    "&:active svg": {
      transform: "scale(1, 1)",
    },
  },
  ({ invert, context }) => {
    if (context.inheritColor) {
      return {
        "&:link,&:visited": {
          color: "inherit",
        },
      };
    }
    if (invert) {
      return {
        "&:link,&:visited": {
          color: colorWhite,
        },
      };
    }
    return {};
  },
  states,
  ({ hasIcon }) => {
    if (hasIcon) {
      return {
        display: "inline-block",
        "& > svg": {
          verticalAlign: "middle",
        },
      };
    }
    return {};
  }
);

const Link = (
  {
    reactRouterLinkComponent,
    invert,
    children,
    forwardedRef,
    icon: Icon,
    iconPosition,
    ...rest
  },
  context
) => {
  if (
    !(reactRouterLinkComponent && rest.to) &&
    (reactRouterLinkComponent || rest.to)
  ) {
    warn(
      "Link",
      "The props `reactRouterLinkComponent` and `to` must be used together."
    );
  }

  const renderChildren = useCallback(() => {
    if (Icon) {
      return (
        <>
          {iconPosition === "left" && (
            <Icon
              color={invert ? "white" : "greyShark"}
              style={{ marginRight: "0.5rem" }}
            />
          )}
          {children}
          {iconPosition === "right" && (
            <Icon
              color={invert ? "white" : "greyShark"}
              style={{ marginLeft: "0.25rem" }}
            />
          )}
        </>
      );
    }
    return children;
  }, [children, Icon, iconPosition, invert]);

  return (
    <StyledLink
      {...safeRest(rest)}
      as={reactRouterLinkComponent || "a"}
      invert={invert}
      context={context}
      ref={forwardedRef}
      hasIcon={!!Icon}
    >
      {renderChildren()}
    </StyledLink>
  );
};
Link.propTypes = {
  reactRouterLinkComponent: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.object,
  ]),
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  href: PropTypes.string,
  invert: PropTypes.bool,
  children: PropTypes.node.isRequired,
  forwardedRef: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  icon: componentWithName("Dependent", true),
  iconPosition: PropTypes.oneOf(["left", "right"]),
};
Link.defaultProps = {
  reactRouterLinkComponent: null,
  to: null,
  href: null,
  invert: undefined,
  forwardedRef: undefined,
  icon: undefined,
  iconPosition: "left",
};

Link.contextTypes = {
  inheritColor: PropTypes.bool,
};

export default withForwardedRef(Link);
