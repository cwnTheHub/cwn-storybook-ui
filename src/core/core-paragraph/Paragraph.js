import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { DependentIconSizeContext, safeRest } from "../../util-helpers";
import * as typography from "../../shared-typography/typography";
import { spacing } from "../../shared-styles";

const paragraphColor = ({ invert }) =>
  invert ? typography.invertedColor : typography.color;
const paragraphInheritColor = ({ inheritColor }) =>
  inheritColor ? { color: "inherit" } : undefined;

const paragraphSize = ({ size }) => typography[size];
const paragraphBold = ({ bold, size }) =>
  bold ? typography.boldFont : typography[`${size}Font`];
const paragraphAlign = ({ align }) => ({
  textAlign: align,
});

export const StyledParagraph = styled.p(
  paragraphColor,
  typography.wordBreak,
  spacing.noSpacing,
  paragraphInheritColor,
  paragraphSize,
  paragraphBold,
  paragraphAlign,
  { sup: typography.sup }
);

const Paragraph = ({ size, invert, children, ...rest }, context) => {
  return (
    <DependentIconSizeContext.Provider value={{ paragraphSize: size, invert }}>
      <StyledParagraph
        {...safeRest(rest)}
        size={size}
        invert={invert}
        inheritColor={context.inheritColor}
      >
        {children}
      </StyledParagraph>
    </DependentIconSizeContext.Provider>
  );
};

Paragraph.propTypes = {
  bold: PropTypes.bool,
  size: PropTypes.oneOf(["small", "medium", "large"]),
  align: PropTypes.oneOf(["left", "center", "right"]),
  invert: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

Paragraph.defaultProps = {
  bold: false,
  size: "medium",
  align: "left",
  invert: false,
};

Paragraph.contextTypes = {
  inheritColor: PropTypes.bool,
};

export default Paragraph;
