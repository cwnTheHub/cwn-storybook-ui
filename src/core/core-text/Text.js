import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { DependentIconSizeContext, safeRest } from "../../util-helpers";
import * as typography from "../../shared-typography/typography";


const textColor = ({ invert }) =>
  invert ? typography.invertedColor : typography.color;
const textInheritColor = ({ inheritColor }) =>
  inheritColor ? { color: "inherit" } : undefined;
const textSize = ({ size }) => typography[size];
const textBold = ({ bold, size }) =>
  bold ? typography.boldFont : typography[`${size}Font`];
const textBlock = ({ block }) => (block ? typography.blockText : undefined);

// This named export is not guaranteed to be maintained and may be removed at any time.
export const StyledText = styled.span(
  textColor,
  textInheritColor,
  textSize,
  textBold,
  textBlock,
  {
    sup: typography.sup,
  }
);

const Text = ({ children, size, invert, ...rest }, context) => (
  <DependentIconSizeContext.Provider value={{ paragraphSize: size, invert }}>
    <StyledText
      {...safeRest(rest)}
      size={size}
      invert={invert}
      inheritColor={context.inheritColor}
    >
      {children}
    </StyledText>
  </DependentIconSizeContext.Provider>
);

Text.propTypes = {
  block: PropTypes.bool,
  bold: PropTypes.bool,
  size: PropTypes.oneOf(["base", "small", "medium", "large"]),
  invert: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

Text.defaultProps = {
  block: false,
  bold: false,
  size: "base",
  invert: false,
};

Text.contextTypes = {
  inheritColor: PropTypes.bool,
};

export default Text;
