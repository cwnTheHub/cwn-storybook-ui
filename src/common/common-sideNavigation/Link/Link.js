import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Text from "../../../core/core-text/Text";
import { safeRest } from "../../../util-helpers";
import {
  colorWhiteLilac,
  colorNemetonPurple,
  colorShuttleGrey,
} from "../../../core/core-colours/colours";
import { warn } from "../../../utils/warn";
import ColoredTextProvider from "../../../utils/components/ColoredTextProvider/ColoredTextProvider";
import Box from "../../../core/core-box/Box";


const activeContainer = {
  borderLeft: `4px solid ${colorNemetonPurple}`,
  color: `${colorNemetonPurple}`,
  backgroundColor: `${colorWhiteLilac}`,
};

const activeText = {
  fontWeight: "bold",
  color: `${colorNemetonPurple}`,
};

const hover = {
  color: `${colorShuttleGrey}`,
  "&:hover": {
    color: `${colorNemetonPurple}`,
  },
};

const BoxContainer = styled(Box)((props) => ({
  "&:hover": {
    backgroundColor: `${colorWhiteLilac}`,
  },
  marginLeft: props.submenulink ? "1rem" : "0",
  borderLeft: props.submenulink ? `4px solid ${colorWhiteLilac}` : "none",
  ...(props.active && !props.submenulink ? activeContainer : undefined),
}));

const BoxWrapper = styled(Box)((props) => ({
  backgroundColor: props.active ? `${colorWhiteLilac}` : "none",
}));

const BoxActive = styled(Box)((props) => ({
  borderLeft:
    props.active && props.submenulink
      ? `4px solid ${colorNemetonPurple}`
      : "none",
  marginLeft: props.submenulink ? "-0.25rem" : "0px",
}));

const StyledTextProvider = styled(ColoredTextProvider)((props) => ({
  ...(props.active ? activeText : hover),
}));

const StyledAnchor = styled.a({
  textDecoration: "none",
});

const Link = ({
  reactRouterLinkComponent,
  children,
  active,
  subMenuLink,
  ...rest
}) => {
  if (
    (reactRouterLinkComponent || rest.to) &&
    !(reactRouterLinkComponent && rest.to)
  ) {
    warn(
      "Link",
      "The props `reactRouterLinkComponent` and `to` must be used together."
    );
  }

  const innerLink = (
    <BoxWrapper active={active ? 1 : 0}>
      <BoxContainer
        vertical={3}
        submenulink={subMenuLink ? 1 : 0}
        active={active ? 1 : 0}
      >
        <BoxActive
          active={active ? 1 : 0}
          horizontal={3}
          submenulink={subMenuLink ? 1 : 0}
        >
          <StyledTextProvider active={active}>
            <Text size={subMenuLink ? "small" : "medium"} bold={active}>
              {children}
            </Text>
          </StyledTextProvider>
        </BoxActive>
      </BoxContainer>
    </BoxWrapper>
  );

  return React.createElement(
    reactRouterLinkComponent || StyledAnchor,
    {
      ...safeRest(rest),
    },
    innerLink
  );
};

Link.propTypes = {
  children: PropTypes.string.isRequired,
  href: PropTypes.string,
  active: PropTypes.bool,
  subMenuLink: PropTypes.bool,
  reactRouterLinkComponent: PropTypes.func,
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

Link.defaultProps = {
  href: undefined,
  active: false,
  subMenuLink: false,
  reactRouterLinkComponent: null,
  to: null,
};

export default Link;
