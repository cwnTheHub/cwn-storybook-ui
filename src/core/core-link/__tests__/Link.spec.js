import React from "react";
import { mount, shallow } from "enzyme";

import { warn } from "../../../utils/warn";
import Link from "../Link";
import Paragraph from "../../core-paragraph/Paragraph";
import Text from "../../core-text/Text";
import { Edit } from "../../core-interactive-icon";
import { StyledDependentSVG } from "../../core-interactive-icon/Dependent";

jest.mock("../../../utils/warn");

describe("Link", () => {
  const doShallow = (overrides = {}) =>
    shallow(<Link {...overrides}>Go home</Link>);
  const doMount = (overrides = {}) => {
    const link = mount(<Link {...overrides}>Some content</Link>);
    return {
      link: link.find(overrides.reactRouterLinkComponent || "a"),
      wrapper: link,
    };
  };
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders", () => {
    const link = doShallow();

    expect(link).toMatchSnapshot();
  });

  it("is an anchor HTML element when using the href attribute", () => {
    const { link } = doMount({ href: "http://nds.com" });

    expect(link.name()).toEqual("a");
    expect(link.props().href).toEqual("http://nds.com");
  });

  it("renders a react router link element when passed as a prop", () => {
    const MyLink = () => <span />;
    const component= doMount({ reactRouterLinkComponent: MyLink });

    expect(component.link.find("MyLink")).toBeTruthy()
  });

  it("must use `reactRouterLinkComponent` and `to` props together", () => {
    const MyLink = () => <span />;
    let link = doMount({ reactRouterLinkComponent: MyLink });

    expect(warn).toHaveBeenCalled();

    link = doShallow({ to: "/about" });

    expect(link.props().to).toBeTruthy();
    expect(warn).toHaveBeenCalled();
  });

  it("can be displayed with the default styles", () => {
    const link = doMount();
    expect(link).toMatchSnapshot();
  });

  it("can be inverted", () => {
    const link = doMount({ invert: true });
    expect(link).toMatchSnapshot();
  });

  it("passes additional attributes to the link element", () => {
    const link = doShallow({ id: "the-link", role: "button" });

    expect(link.props().id).toEqual( "the-link");
    expect(link.props().role).toEqual("button");
  });

  it("does not allow custom CSS", () => {
    const link = doMount({
      className: "my-custom-class",
      style: { color: "hotpink" },
    });

    expect(link.link.hasClass("my-custom-class")).toEqual(false);
    expect(link.link.props().style).toBeFalsy();
  });

  

  describe("dependent icons", () => {
    it("displays a Dependent Icon", () => {
      const link = doMount({ id: "the-link", role: "button", icon: Edit }).link;
      expect(link.find('[data-testid="dependentSvg"]')).toBeTruthy();
    });

    it("displays a Dependent Icon and sets alignment", () => {
      const linkLeft = doMount({
        id: "the-link",
        role: "button",
        icon: Edit,
        iconPosition: "right",
      }).link;
      const linkRight = doMount({
        id: "the-link",
        role: "button",
        icon: Edit,
        iconPosition: "right",
      }).link;

      expect(linkLeft).toMatchSnapshot();
      expect(linkRight).toMatchSnapshot();
    });

    it("inherits size from paragraph", () => {
      const link = mount(
        <Paragraph size="large">
          <Link href="#" icon={Edit}>
            Edit
          </Link>
        </Paragraph>
      );

      const size = link.find(StyledDependentSVG).prop("paragraphSize");
      expect(size).toEqual("large");
    });

    it("inherits size from text", () => {
      const link = mount(
        <Text size="small">
          <Link href="#" icon={Edit}>
            Edit
          </Link>
        </Text>
      );

      const size = link.find(StyledDependentSVG).prop("paragraphSize");
      expect(size).toEqual("small");
    });

    it("inherits size from nearest context", () => {
      const link = mount(
        <Paragraph size="small">
          Need to make changes?
          <Text size="large">
            <Link href="#" icon={Edit}>
              Edit
            </Link>
          </Text>
        </Paragraph>
      );

      const size = link.find(StyledDependentSVG).prop("paragraphSize");
      expect(size).toEqual("large");
    });
  });
});
