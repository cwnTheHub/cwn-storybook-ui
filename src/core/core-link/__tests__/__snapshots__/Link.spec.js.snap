// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link can be displayed with the default styles 1`] = `
Object {
  "link": <a
    className="sc-aXZVg jPsNzi"
    href={null}
    to={null}
  >
    Some content
  </a>,
  "wrapper": <WithForwardedRef(Link)>
    <Link
      forwardedRef={null}
      href={null}
      iconPosition="left"
      reactRouterLinkComponent={null}
      to={null}
    >
      <styled.a
        as="a"
        context={
          Object {
            "inheritColor": undefined,
          }
        }
        hasIcon={false}
        href={null}
        to={null}
      >
        <a
          className="sc-aXZVg jPsNzi"
          href={null}
          to={null}
        >
          Some content
        </a>
      </styled.a>
    </Link>
  </WithForwardedRef(Link)>,
}
`;

exports[`Link can be inverted 1`] = `
Object {
  "link": <a
    className="sc-aXZVg bkAQon"
    href={null}
    to={null}
  >
    Some content
  </a>,
  "wrapper": <WithForwardedRef(Link)
    invert={true}
  >
    <Link
      forwardedRef={null}
      href={null}
      iconPosition="left"
      invert={true}
      reactRouterLinkComponent={null}
      to={null}
    >
      <styled.a
        as="a"
        context={
          Object {
            "inheritColor": undefined,
          }
        }
        hasIcon={false}
        href={null}
        invert={true}
        to={null}
      >
        <a
          className="sc-aXZVg bkAQon"
          href={null}
          to={null}
        >
          Some content
        </a>
      </styled.a>
    </Link>
  </WithForwardedRef(Link)>,
}
`;

exports[`Link dependent icons displays a Dependent Icon and sets alignment 1`] = `
<a
  className="sc-aXZVg dImOAW"
  href={null}
  id="the-link"
  role="button"
  to={null}
>
  Some content
  <Dependent
    color="greyShark"
    style={
      Object {
        "marginLeft": "0.25rem",
      }
    }
  >
    <Dependent
      color="greyShark"
      style={
        Object {
          "marginLeft": "0.25rem",
        }
      }
    >
      <styled.svg
        color="greyShark"
        data-testid="dependentSvg"
        height="24"
        style={
          Object {
            "marginLeft": "0.25rem",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <svg
          aria-hidden={true}
          className="sc-kpDqfm hJvIhZ"
          color="greyShark"
          data-testid="dependentSvg"
          focusable={false}
          height="24"
          style={
            Object {
              "marginLeft": "0.25rem",
            }
          }
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M19.717 2c.612 0 1.198.235 1.61.646a2.275 2.275 0 01.135 3.086l-.138.148L7.11 20.093a2.215 2.215 0 01-.227.205l-.03.015-4.114 1.648a.544.544 0 01-.581-.119.536.536 0 01-.146-.495l.027-.087 1.644-4.118a.519.519 0 01.07-.124l.05-.056L18.107 2.646A2.281 2.281 0 0119.718 2zm-15.1 15.715l-1.11 2.776 2.772-1.11-1.663-1.666zM16.114 6.167L5.358 16.931l1.68 1.682 1.335-1.37c1.2-1.216 3.012-3.027 5.672-5.67l3.753-3.721-1.683-1.685zm4.454-2.753a1.222 1.222 0 00-1.595-.096l-.103.091-1.993 1.995 1.686 1.689 2.002-1.982c.23-.224.355-.524.355-.844a1.19 1.19 0 00-.352-.853z"
            fillRule="evenodd"
          />
        </svg>
      </styled.svg>
    </Dependent>
  </Dependent>
</a>
`;

exports[`Link dependent icons displays a Dependent Icon and sets alignment 2`] = `
<a
  className="sc-aXZVg dImOAW"
  href={null}
  id="the-link"
  role="button"
  to={null}
>
  Some content
  <Dependent
    color="greyShark"
    style={
      Object {
        "marginLeft": "0.25rem",
      }
    }
  >
    <Dependent
      color="greyShark"
      style={
        Object {
          "marginLeft": "0.25rem",
        }
      }
    >
      <styled.svg
        color="greyShark"
        data-testid="dependentSvg"
        height="24"
        style={
          Object {
            "marginLeft": "0.25rem",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <svg
          aria-hidden={true}
          className="sc-kpDqfm hJvIhZ"
          color="greyShark"
          data-testid="dependentSvg"
          focusable={false}
          height="24"
          style={
            Object {
              "marginLeft": "0.25rem",
            }
          }
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M19.717 2c.612 0 1.198.235 1.61.646a2.275 2.275 0 01.135 3.086l-.138.148L7.11 20.093a2.215 2.215 0 01-.227.205l-.03.015-4.114 1.648a.544.544 0 01-.581-.119.536.536 0 01-.146-.495l.027-.087 1.644-4.118a.519.519 0 01.07-.124l.05-.056L18.107 2.646A2.281 2.281 0 0119.718 2zm-15.1 15.715l-1.11 2.776 2.772-1.11-1.663-1.666zM16.114 6.167L5.358 16.931l1.68 1.682 1.335-1.37c1.2-1.216 3.012-3.027 5.672-5.67l3.753-3.721-1.683-1.685zm4.454-2.753a1.222 1.222 0 00-1.595-.096l-.103.091-1.993 1.995 1.686 1.689 2.002-1.982c.23-.224.355-.524.355-.844a1.19 1.19 0 00-.352-.853z"
            fillRule="evenodd"
          />
        </svg>
      </styled.svg>
    </Dependent>
  </Dependent>
</a>
`;

exports[`Link renders 1`] = `
<Link
  forwardedRef={null}
  href={null}
  iconPosition="left"
  reactRouterLinkComponent={null}
  to={null}
>
  Go home
</Link>
`;
